protocols/mp-ecdsa/src/client.rs

```diff
@@ -0,0 +1,241 @@
+    client: Arc<C>,
+    key_store: DKGKeystore,
+    finality_notification_stream: Arc<Mutex<FinalityNotifications<B>>>,
+    block_import_notification_stream: Arc<Mutex<ImportNotifications<B>>>,
+    latest_finality_notification: Arc<parking_lot::Mutex<Option<FinalityNotification<B>>>>,
+    logger: DebugLogger,
+    _block: std::marker::PhantomData<BE>,
+}
+
+pub async fn create_client<B: Block, BE: Backend<B>>(
+    _config: &MpEcdsaProtocolConfig,
+) -> Result<MpEcdsaClient<B, BE, ()>, Box<dyn Error>> {
+    Ok(MpEcdsaClient {
+        _block: std::marker::PhantomData,
+    })
+}
+
+pub trait ClientWithApi<B, BE>:
+    BlockchainEvents<B> + HeaderBackend<B> + ProvideRuntimeApi<B> + Send + Sync
+where
+    B: Block,
+    BE: Backend<B>,
+    <Self as ProvideRuntimeApi<B>>::Api:
+        DKGApi<B, AuthorityId, NumberFor<B>, MaxProposalLength, MaxAuthorities>,
+{
+}
+
+impl<B, BE, T> ClientWithApi<B, BE> for T
+where
+    B: Block,
+    BE: Backend<B>,
+    T: BlockchainEvents<B> + HeaderBackend<B> + ProvideRuntimeApi<B> + Send + Sync,
+    <T as ProvideRuntimeApi<B>>::Api:
+        DKGApi<B, AuthorityId, NumberFor<B>, MaxProposalLength, MaxAuthorities>,
+{
+}
+
+impl<B: Block, BE: Backend<B>, C: ClientWithApi<B, BE>> MpEcdsaClient<B, BE, C> {
+    pub async fn validator_set(
+        &self,
+        header: &B::Header,
+    ) -> Option<(
+        AuthoritySet<AuthorityId, MaxAuthorities>,
+        AuthoritySet<AuthorityId, MaxAuthorities>,
+    )> {
+        let new = if let Some((new, queued)) = find_authorities_change::<B>(header) {
+            Some((new, queued))
+        } else {
+            let at = header.hash();
+            let current_authority_set = exec_client_function(&self.client, move |client| {
+                client.runtime_api().authority_set(at).ok()
+            })
+            .await;
+
+            let queued_authority_set = exec_client_function(&self.client, move |client| {
+                client.runtime_api().queued_authority_set(at).ok()
+            })
+            .await;
+
+            match (current_authority_set, queued_authority_set) {
+                (Some(current), Some(queued)) => Some((current, queued)),
+                _ => None,
+            }
+        };
+
+        self.logger
+            .trace(format!("🕸️  active validator set: {new:?}"));
+
+        new
+    }
+
+    pub async fn should_execute_new_keygen(
+        &self,
+        header: &B::Header,
+    ) -> AnticipatedKeygenExecutionStatus {
+        // query runtime api to check if we should execute new keygen.
+        let at = header.hash();
+        let (execute, force_execute) = self
+            .exec_client_function(move |client| {
+                client.runtime_api().should_execute_new_keygen(at).unwrap_or_default()
+            })
+            .await;
+
+        AnticipatedKeygenExecutionStatus { execute, force_execute }
+    }
+
+    /// Get the next DKG public key
+    pub async fn get_next_dkg_pub_key(
```

Comment:
There is no next public key anymore, there is just a Phase 1 Job that indicates completing a DKG w/ specified configuration parameters such as validator participants.